src/utils/workflow_utils/src/workflow_utils.c

```diff
@@ -2291,36 +2291,42 @@ ADUC_Result workflow_create_from_inline_step(ADUC_WorkflowHandle base, int stepI
     // Keep only file needed by this step entry. Remove the rest.
     JSON_Array* stepFiles = json_object_get_array(stepObject, ADUCITF_FIELDNAME_FILES);
     JSON_Object* baseFiles = json_object_get_object(updateManifestObject, ADUCITF_FIELDNAME_FILES);
-    int fileCount = (int)json_object_get_count(baseFiles);
-    for (int b = fileCount - 1; b >= 0; b--)
-    {
-        const char* baseFileId = json_object_get_name(baseFiles, b);
+
+    size_t fileCount = json_object_get_count(baseFiles);
+    do {
+        fileCount--;
+
+        const char* baseFileId = json_object_get_name(baseFiles, fileCount);
 
         // If file is in the instruction, merge their properties.
         bool fileRequired = false;
-        int stepFilesCount = (int)json_array_get_count(stepFiles);
-        for (int i = stepFilesCount - 1; i >= 0; i--)
-        {
+
+        size_t stepFilesCount = json_array_get_count(stepFiles);
+        do{
+            stepFilesCount--;
```

Comment:
same edge case here. Should have >= 1 steps but should handle case where there are zero steps and return a failure resultCode

Reply:
This isn't within the scope of the PR but I'll create a bug to track in case it is needed.